module_test.go

```diff
@@ -420,4 +498,142 @@ func TestModuleFailures(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("error in Provide shows logs in module", func(t *testing.T) {
+		t.Parallel()
+
+		var spy fxlog.Spy
+
+		redis := fx.Module("redis",
+			fx.Provide(func() string {
+				return "redis"
+			}),
+			fx.Supply(&spy),
+			fx.WithLogger(func(spy *fxlog.Spy) fxevent.Logger {
+				return spy
+			}),
+			fx.Provide(&bytes.Buffer{}), // not passing in a constructor
+			fx.Invoke(func(s string) {
+				assert.Fail(t, "this should never run")
+			}),
+		)
+
+		app := fx.New(
+			redis,
+			fx.Invoke(func(s string) {
+				assert.Fail(t, "this should never run")
+			}),
+		)
+
+		err := app.Err()
+		require.Error(t, err)
+		assert.Contains(t,
+			err.Error(),
+			"must provide constructor function, got  (type *bytes.Buffer)",
+		)
+
+		assert.Equal(t,
+			[]string{"Provided", "Supplied", "Provided", "LoggerInitialized"},
+			spy.EventTypes(),
+		)
+	})
+
+	t.Run("logger in module failed to build", func(t *testing.T) {
+		t.Parallel()
+
+		redis := fx.Module("redis",
+			fx.Provide(func() string {
+				return "redis"
+			}),
+			fx.WithLogger(func() (fxevent.Logger, error) {
+				return nil, errors.New("error building logger")
+			}),
+			fx.Invoke(func(s string) {
+				assert.Fail(t, "this should never run")
+			}),
+		)
+
+		var buff bytes.Buffer
+		app := fx.New(
+			redis,
+			fx.Logger(log.New(&buff, "", 0)),
+			fx.Invoke(func(s string) {
+				assert.Fail(t, "this should never run")
+			}),
+		)
+
+		err := app.Err()
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "error building logger")
+
+		out := buff.String()
+		assert.Contains(t, out, "[Fx] ERROR\t\tFailed to initialize custom logger")
+	})
+
+	t.Run("logger dependency in module failed to build", func(t *testing.T) {
+		t.Parallel()
+
+		redis := fx.Module("redis",
+			fx.Provide(func() string {
+				return "redis"
+			}),
+			fx.Provide(func() (*zap.Logger, error) {
+				return nil, errors.New("error building logger dependency")
+			}),
+			fx.WithLogger(func(log *zap.Logger) fxevent.Logger {
+				t.Errorf("WithLogger must not be called")
+				panic("must not be called")
+			}),
+			fx.Invoke(func(s string) {
+				assert.Fail(t, "this should never run")
+			}),
+		)
+
+		var buff bytes.Buffer
+		app := fx.New(
+			redis,
+			fx.Logger(log.New(&buff, "", 0)),
```

Comment:
this is deprecated - can you change these with `WithLogger` and use spy like the other tests?